### PR TITLE
Janitor cart

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Adds janitor cart to codebase.

Jani cart has behavior of mop bucket, item storage and item slots The main challenge here was not creating any new components exclusive for this bucket on wheels. And for my surprise its kinda worked. Some required changes was:
- Storages now has blacklist. Storage popup "Can't insert" now optional.
- Item Mapper now works not only with storages, but with any containers in general.
- Item Mapper now has min and max count of item. If there is smaller or bigger amount items inside containers, sprite layer won't be shown. Useful for stack like behavior. 

Some known issues / weird UX:
- In SS13 you could place trash directly into trash bag attached to cart. Right now it doesn't work this way.
- You can insert or eject mop, only by verbs. In SS13 there was clunky system where first click wet mop and second click place mop inside. With current mopping system it can't be done same way.

**Screenshots**

https://user-images.githubusercontent.com/6161335/161146969-d8a80e4d-f2a8-4bad-bedf-78c45a9ef5a8.mp4

**Changelog**

:cl:
- add: Added  janitor cart to codebase. Not mapped yet.
- tweak: Wet floor sign and mop doesn't fit in pockets now.

### CODE REVIEW

**Comment by Elijahrane** [comment]
There was a discussion in #development in Discord about how having 2 different things named "Janicart" (which is being added by the vehicle PR) and "Janitor Cart" is a bit confusing. Could you maybe change the display name to "mop bucket cart" or "mop trolley" or "janitor trolley" or something similar?

**Comment by Macoron** [comment]
> There was a discussion in #development in Discord about how having 2 different things named "Janicart" (which is being added by the vehicle PR) and "Janitor Cart" is a bit confusing, even if it is that way on /tg/. Could you maybe change the display name to "mop bucket cart" or "mop trolley" or "janitor trolley" or something similar?

I renamed it to "janitorial trolley"

File: Content.Shared/Containers/ItemSlot/ItemSlotsComponent.cs [hunk 1]
```diff
@@ -116,6 +116,12 @@ public sealed class ItemSlot
         [ViewVariables(VVAccess.ReadWrite)]
         public bool Locked = false;
 
+        /// <summary>
+        ///     Whether the item slots system will attempt to insert this item to the user's hands when interacted with.
+        /// </summary>
+        [DataField("insertOnInteract")]
+        public bool InsertOnInteract = true;
+
```
Comment (mirrorcult) [inline Content.Shared/Containers/ItemSlot/ItemSlotsComponent.cs:124]: seems like this also blocks inserting items into the slot, not just ejecting to the hands as the comment claims

File: Content.Server/Storage/Components/ServerStorageComponent.cs [hunk 2]
```diff
@@ -71,6 +71,11 @@ public sealed class ServerStorageComponent : SharedStorageComponent, IInteractUs
 
         [DataField("whitelist")]
         private EntityWhitelist? _whitelist = null;
+        [DataField("blacklist")]
+        public EntityWhitelist? Blacklist = null;
+
+        [DataField("popup")]
+        public bool ShowPopup = true;
```
Comment (mirrorcult) [inline Content.Server/Storage/Components/ServerStorageComponent.cs:78]: xml docs would be nice for this

File: Resources/Prototypes/Entities/Objects/Specific/Janitorial/janitor.yml [hunk 3]
```diff
@@ -74,3 +74,128 @@
...
+      thresholds:
+        - trigger:
+            !type:DamageTrigger
+            damage: 200
+          behaviors:
+            - !type:DoActsBehavior
+              acts: ["Destruction"]
+            - !type:PlaySoundBehavior
+              sound:
+                path: /Audio/Effects/metalbreak.ogg
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Specific/Janitorial/janitor.yml:155]: presumably this should also empty all containers

**Review by mirrorcult — CHANGES_REQUESTED** [review]
yaml supremacy

File: Content.Shared/Containers/ItemSlot/ItemSlotsComponent.cs (hunk 1 above)
Comment (Macoron) [reply Content.Shared/Containers/ItemSlot/ItemSlotsComponent.cs]: Yeah, comment wasn't correct. Should be ok now

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**